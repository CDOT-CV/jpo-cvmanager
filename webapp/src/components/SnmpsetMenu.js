import React from 'react'
import { useDispatch, useSelector } from 'react-redux'

import {
  selectChangeSuccess,
  selectErrorState,
  selectDestIp,
  selectSnmpMsgType,
  selectSnmpFilterMsg,
  selectSnmpFilterErr,

  // Actions
  submitSnmpSet,
  deleteSnmpSet,
  filterSnmp,
  setDestIp,
  setMsgType,
} from '../generalSlices/configSlice'

import { selectRsuIpv4, selectRsuManufacturer } from '../generalSlices/rsuSlice'

import './css/SnmpwalkMenu.css'

const SnmpsetMenu = ({ type, rsuIpList }) => {
  const dispatch = useDispatch()

  const changeSuccess = useSelector(selectChangeSuccess)
  const errorState = useSelector(selectErrorState)
  const destIp = useSelector(selectDestIp)
  const snmpMsgType = useSelector(selectSnmpMsgType)
  const snmpFilterMsg = useSelector(selectSnmpFilterMsg)
  const snmpFilterErr = useSelector(selectSnmpFilterErr)

  const rsuIp = useSelector(selectRsuIpv4)
  const rsuManufacturer = useSelector(selectRsuManufacturer)

  return (
    <div id="snmpdiv">
      <h2 id="snmpheader">Message Forwarding</h2>
      <form id="snmpform">
        <label id="snmplabel">
          <strong>Destination IP:</strong>
          <input id="snmpinput" type="text" value={destIp} onChange={(e) => dispatch(setDestIp(e.target.value))} />
        </label>
        <label id="snmplabel">
          <strong>Message Type:</strong>
          <select id="snmpdropdown" value={snmpMsgType} onChange={(e) => dispatch(setMsgType(e.target.value))}>
            <option value="bsm">BSM</option>
            <option value="spat">SPaT</option>
            <option value="map">MAP</option>
            <option value="srm">SRM</option>
            <option value="ssm">SSM</option>
            <option value="tim">TIM</option>
            <option value="psm">PSM</option>
          </select>
        </label>
      </form>

      <button id="refreshbtn" onClick={() => dispatch(submitSnmpSet(rsuIpList))}>
        Add Forwarding
      </button>
      {type !== 'single_rsu' && (
        <button
          id="refreshbtn"
          onClick={() =>
            dispatch(
              deleteSnmpSet({
                ipList: rsuIpList,
                snmpMsgType: snmpMsgType,
                destIp: destIp,
              })
            )
          }
        >
          Delete Forwarding
        </button>
      )}

      {changeSuccess ? (
        <div>
          <p id="successtext">Successful write to RSU</p>
        </div>
      ) : (
        <p id="infotext"></p>
      )}
      {errorState !== '' ? <p id="warningtext">{errorState}</p> : <div />}

      {type !== 'single_rsu' ? (
        <div>
          <p id="snmpfiltertext" margintop="40px">
            By specifying a destination IP address along with a message type, you can add message forwarding to any of
            the selected RSUs or delete message forwarding from any RSUs that already have that configuration.
          </p>
        </div>
      ) : (
        <div />
      )}

      {rsuManufacturer === 'Yunex' ? (
        <div>
          <p id="snmpfiltertext" margintop="40px">
            Yunex RSUs use different SNMP tables for message TX and RX forwarding. <br /> BSM and SSM are on the RX
            table. MAP, SPaT and SRM are on the TX table. <br /> Start over from the 1 index for each table.
          </p>
        </div>
      ) : (
        <div />
      )}

      {rsuManufacturer === 'Commsignia' ? (
        <div>
          <p id="snmpfiltertext" margintop="40px">
            If you are configuring SPaT or MAP forwarding, apply the TX message <br /> filter after your configuration
            has been applied
          </p>
          <button id="refreshbtn" onClick={() => dispatch(filterSnmp([rsuIp]))}>
            Apply TX Filter
          </button>
          {snmpFilterMsg !== '' ? (
            <div>
              {snmpFilterErr === true ? (
                <p id="warningtext">{snmpFilterMsg}</p>
              ) : (
                <p id="successtext">{snmpFilterMsg}</p>
              )}
            </div>
          ) : (
            <div />
          )}
        </div>
      ) : (
        <div />
      )}
    </div>
  )
}

export default SnmpsetMenu
