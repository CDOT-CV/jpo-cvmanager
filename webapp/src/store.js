import { configureStore } from "@reduxjs/toolkit";
import rsuReducer from "./slices/rsuSlice";
import userReducer from "./slices/userSlice";
import wzdxReducer from "./slices/wzdxSlice";
import configReducer from "./slices/configSlice";

const store = configureStore({
  reducer: {
    rsu: rsuReducer,
    user: userReducer,
    wzdx: wzdxReducer,
    config: configReducer,
  },
});

export default store;
