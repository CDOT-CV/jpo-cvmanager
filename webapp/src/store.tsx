import { Action, ThunkAction, configureStore } from '@reduxjs/toolkit'
import rsuReducer from './generalSlices/rsuSlice'
import userReducer from './generalSlices/userSlice'
import wzdxReducer from './generalSlices/wzdxSlice'
import configReducer from './generalSlices/configSlice'
import adminAddOrganizationReducer from './features/adminAddOrganization/adminAddOrganizationSlice'
import adminAddRsuReducer from './features/adminAddRsu/adminAddRsuSlice'
import adminAddUserReducer from './features/adminAdduser/adminAddUserSlice'
import adminEditOrganizationReducer from './features/adminEditOrganization/adminEditOrganizationSlice'
import adminEditRsuReducer from './features/adminEditRsu/adminEditRsuSlice'
import adminEditUserReducer from './features/adminEditUser/adminEditUserSlice'
import adminOrganizationTabReducer from './features/adminOrganizationTab/adminOrganizationTabSlice'
import adminOrganizationTabUserReducer from './features/adminOrganizationTabUser/adminOrganizationTabUserSlice'
import adminOrganizationTabRsuReducer from './features/adminOrganizationTabRsu/adminOrganizationTabRsuSlice'
import adminRsuTabReducer from './features/adminRsuTab/adminRsuTabSlice'
import adminUserTabReducer from './features/adminUserTab/adminUserTabSlice'
import menuReducer from './features/menu/menuSlice'
import intersectionMapReducer from './components/map/map-slice'
import intersectionMapLayerStyleReducer from './components/map/map-layer-style-slice'

export const setupStore = (preloadedState: any) => {
  return configureStore({
    reducer: {
      rsu: rsuReducer,
      user: userReducer,
      wzdx: wzdxReducer,
      config: configReducer,
      adminAddOrganization: adminAddOrganizationReducer,
      adminAddRsu: adminAddRsuReducer,
      adminAddUser: adminAddUserReducer,
      adminEditOrganization: adminEditOrganizationReducer,
      adminEditRsu: adminEditRsuReducer,
      adminEditUser: adminEditUserReducer,
      adminOrganizationTab: adminOrganizationTabReducer,
      adminOrganizationTabUser: adminOrganizationTabUserReducer,
      adminOrganizationTabRsu: adminOrganizationTabRsuReducer,
      adminRsuTab: adminRsuTabReducer,
      adminUserTab: adminUserTabReducer,
      menu: menuReducer,
      intersectionMap: intersectionMapReducer,
      intersectionMapLayerStyle: intersectionMapLayerStyleReducer,
    },
    preloadedState,
  })
}

type AppStore = ReturnType<typeof setupStore>
export type AppState = ReturnType<AppStore['getState']>
type AppThunk<ReturnType = void> = ThunkAction<ReturnType, AppState, unknown, Action>

export type RootState = ReturnType<ReturnType<typeof setupStore>['getState']>
