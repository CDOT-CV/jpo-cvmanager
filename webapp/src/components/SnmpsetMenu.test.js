import React from "react";
import { render } from "@testing-library/react";
import SnmpsetMenu from "./SnmpsetMenu";
import { Provider } from "react-redux";
import { setupStore } from "../store";

it("should take a snapshot", () => {
  const { asFragment } = render(
    <Provider store={setupStore({})}>
      <SnmpsetMenu />
    </Provider>
  );

  expect(
    asFragment(
      <Provider store={setupStore({})}>
        <SnmpsetMenu />
      </Provider>
    )
  ).toMatchSnapshot();
});
