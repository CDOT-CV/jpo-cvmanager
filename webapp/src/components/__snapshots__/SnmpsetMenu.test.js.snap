// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<div>
  <div
    id="snmpdiv"
  >
    <h2
      id="snmpheader"
    >
      Message Forwarding
    </h2>
    <form
      id="snmpform"
    >
      <label
        id="snmplabel"
      >
        <strong>
          Destination IP:
        </strong>
        <input
          id="snmpinput"
          type="text"
          value=""
        />
      </label>
      <label
        id="snmplabel"
      >
        <strong>
          Message Type:
        </strong>
        <select
          id="snmpdropdown"
        >
          <option
            value="bsm"
          >
            BSM
          </option>
          <option
            value="spat"
          >
            SPaT
          </option>
          <option
            value="map"
          >
            MAP
          </option>
          <option
            value="srm"
          >
            SRM
          </option>
          <option
            value="ssm"
          >
            SSM
          </option>
        </select>
      </label>
    </form>
    <button
      id="refreshbtn"
    >
      Add Forwarding
    </button>
    <button
      id="refreshbtn"
    >
      Delete Forwarding
    </button>
    <p
      id="infotext"
    />
    <div />
    <div />
    <div />
  </div>
</div>
`;
